ape_foundry/provider.py

```diff
@@ -623,4 +624,12 @@ def reset_fork(self, block_number: Optional[int] = None):
         if block_number is not None:
             forking_params["blockNumber"] = block_number
 
-        return self._make_request("anvil_reset", [{"forking": forking_params}])
+        result = self._make_request("anvil_reset", [{"forking": forking_params}])
+
+        try:
+            # reset next block base fee to that of new chain head if can
+            self._make_request("anvil_setNextBlockBaseFeePerGas", [self.base_fee])
+        except APINotImplementedError:
```

Comment:
Is this for like older Anvil versions or something?
Maybe it would be nice to warn here, `logger.warning("`anvil_setNextBlockBaseFeePerGas` RPC not found - base fee may not be reset.")`

Reply:
Ohhh is it `base_fee` that causes it?  I thought it `_make_request("anvil_setNextBlockBaseFeePerGas"`

Perhaps it would be best to isolate the `try/except` to only the part we worry about.

Reply:
ah yea that's confusing sorry. will update.

Reply:
also, some fork tests that don't use reset_fork weren't passing on my local when I tried to run.

saw the ignore condition in gh actions, but wasn't sure if failing locally for tests not relevant to reset_fork changes made sense?

Reply:
Those should pass actually! What are the errors? I could help.

We ignore in CI because we can't share API keys with forks

Reply:
Ah, that's my bad then! :( ... They weren't passing locally even prior to changes made, so not really sure why or where I'm going wrong? What I see on my end when running

```
$ pytest -m fork
```

is always the same gas estimation error like in the summary below:

```
---------- coverage: platform darwin, python 3.9.6-final-0 -----------
Name                        Stmts   Miss Branch BrPart  Cover
-------------------------------------------------------------
ape_foundry/__init__.py        15      0      2      0   100%
ape_foundry/constants.py        1      0      0      0   100%
ape_foundry/exceptions.py       6      1      0      0    83%
ape_foundry/provider.py       390    126    135     35    60%
ape_foundry/version.py          2      2      0      0     0%
-------------------------------------------------------------
TOTAL                         414    129    137     35    61%
Coverage HTML written to dir htmlcov
Coverage XML written to file coverage.xml

================================ short test summary info ================================
FAILED tests/test_fork_provider.py::test_goerli_impersonate - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
FAILED tests/test_fork_provider.py::test_mainnet_impersonate - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
FAILED tests/test_fork_provider.py::test_transaction_unknown_contract_as_sender - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
ERROR tests/test_fork_provider.py::test_transaction[solidity] - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
ERROR tests/test_fork_provider.py::test_transaction[vyper] - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
ERROR tests/test_fork_provider.py::test_revert[solidity] - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
ERROR tests/test_fork_provider.py::test_revert[vyper] - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
ERROR tests/test_fork_provider.py::test_contract_revert_no_message[solidity] - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
ERROR tests/test_fork_provider.py::test_contract_revert_no_message[vyper] - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
ERROR tests/test_fork_provider.py::test_transaction_contract_as_sender[solidity] - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
ERROR tests/test_fork_provider.py::test_transaction_contract_as_sender[vyper] - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
ERROR tests/test_fork_provider.py::test_get_receipt[solidity] - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
ERROR tests/test_fork_provider.py::test_get_receipt[vyper] - ape.exceptions.TransactionError: Gas estimation failed: 'invalid type: number, expec...
=========== 3 failed, 7 passed, 38 deselected, 1 warning, 10 errors in 27.95s ===========
```

Reply:
Ah no worries! I fixed that in the last PR. I just updated your branch with `main`. I ran `pytest -m fork` and only 2 tests fail now. However, they are unrelated to this PR and I fixed them here: https://github.com/ApeWorX/ape-foundry/pull/34

So you are good! Thank you: :)

Reply:
So last optional TODO is isolate the `try/except` to only include the `self.base_fee` part

Reply:
sounds good! gonna fix this now, thank you 🫡

Reply:
should be good to go know I think! but lemme know if not.